httpd-true

```diff
@@ -27,12 +27,17 @@ class RequestHandler(http.server.BaseHTTPRequestHandler):
 
 class SockInheritHTTPServer(http.server.HTTPServer):
     def __init__(self, address_info, handler, bind_and_activate=True):
-        self.socket = get_systemd_socket()
         # Note that we call it with bind_and_activate = False.
         http.server.HTTPServer.__init__(self,
```

Comment:
Right, now I've had my coffee and went looking at https://hg.python.org/cpython/file/3.5/Lib/socketserver.py#l433 and I see the issue!



So yes, we're not reusing the socket as we should. Merging this.